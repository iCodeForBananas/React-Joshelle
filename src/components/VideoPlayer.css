.VideoPlayer-container {
  width: 500px;
  height: 360px;
  top: 100px;
  left: 100px;
  position: absolute;
  z-index: 3;
}

.VideoPlayer-tv {
  background: url(/tv.png) no-repeat center;
  background-size: contain;
  z-index: 10;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  position: absolute;
}

.VideoPlayer-spin {
  animation: Video-spin infinite 0.421s ease;
}

@keyframes Video-spin {
  50% {
    transform: scale(1.3, 1.3) rotate(5deg);
  }
  100% {
    transform: scale(1, 1) rotate(-5deg);
  }
}

.VideoPlayer-iframe {
  position: relative;
  z-index: 1;
  position: relative;
  z-index: 1;
  margin: 20px 0 0 20px;
}

@media only screen and (max-width: 440px) {
  .VideoPlayer-container {
    top: 40px;
    left: -50px;
    height: 200px;
  }

  .VideoPlayer-container video {
    height: 160px;
  }
}
