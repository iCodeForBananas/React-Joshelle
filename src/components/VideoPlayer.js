// @flow
import React from "react";
import "./VideoPlayer.css";

type Props = {};
type State = {};

class VideoPlayer extends React.Component<Props, State> {
  componentDidMount() {
    const videoPlayerEl: HTMLVideoElement = document.getElementById("VideoPlayer-video");
    videoPlayerEl && videoPlayerEl.play();
  }

  render() {
    return (
      <div className="VideoPlayer-spin VideoPlayer-container">
        <div className="VideoPlayer-tv" />
        <video
          id="VideoPlayer-video"
          autoPlay={true}
          width="400"
          height="300"
          className="VideoPlayer-iframe"
          loop={true}
        >
          <source src="/heyheyhey.mp4" />
        </video>
      </div>
    );
  }
}

export default VideoPlayer;
