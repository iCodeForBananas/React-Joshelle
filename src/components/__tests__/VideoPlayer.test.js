import React from "react";
import VideoPlayer from "../VideoPlayer";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<VideoPlayer />).toJSON();
  expect(tree).toMatchSnapshot();
});
