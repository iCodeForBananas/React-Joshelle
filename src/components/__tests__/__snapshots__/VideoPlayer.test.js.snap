// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="VideoPlayer-spin VideoPlayer-container"
>
  <div
    className="VideoPlayer-tv"
  />
  <video
    autoPlay={true}
    className="VideoPlayer-iframe"
    height="300"
    id="VideoPlayer-video"
    loop={true}
    width="400"
  >
    <source
      src="/heyheyhey.mp4"
    />
  </video>
</div>
`;
